src/use-case/my-script.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+# BASH3 Boilerplate
+#
+# This file:
+#
+#  - Is a template to write better bash scripts
+#  - Is delete-key friendly, in case you don't need e.g. command line option parsing
+#
+# More info:
+#
+#  - https://github.com/kvz/bash3boilerplate
+#  - http://kvz.io/blog/2013/02/26/introducing-bash3boilerplate/
+#
+# Version: 2.0.0
+#
+# Authors:
+#
+#  - Kevin van Zonneveld (http://kvz.io)
+#  - Izaak Beekman (https://izaakbeekman.com/)
+#  - Alexander Rathai (Alexander.Rathai@gmail.com)
+#  - Dr. Damian Rouson (http://www.sourceryinstitute.org/) (documentation)
+#
+# Usage:
+#
+#  LOG_LEVEL=7 ./main.sh -f /tmp/x -d
+#
+# Licensed under MIT
+# Copyright (c) 2013 Kevin van Zonneveld (http://kvz.io)
+
+# Incorporate the bash3boilerplate content that does not need to change from one use to the next
+# and designate the name of the file containing the current file's usage as the file with "-usage"
+# appended to the current file's name:
+# name followed by "-usage"
+__filename="$(basename "${BASH_SOURCE[0]}")"
+. bootstrap.sh -f ${__filename}-usage
+
+# Command-Line Interface (CLI) 
+# ----------------------------
+# This 'while' block below reads the file named in usage_page, which will also be used to 
+# parse CLI options, their associated arguments, their default values, and their status as
+# required or optional.  See the above-referenced bootstrap script for a comprehensive 
+# example of usage page contents.  See my-script.sh-usage for the specific usage page for
+# the current script.
+while IFS='' read -r line || [[ -n "$line" ]]; do
+  usage="${usage:-}""${line}"$'\n'
```

Comment:
There was one point in the process of doing this refactoring when the usage lines read by my-script.sh were being appended to the usage lines from bootstrap.sh, but I found a way to overwrite instead of append.  I don't recall the exact change that switched the behavior as I was moving fast and it was quite late in the evening, but it works now as intended. 

The number one goal here is to encourage modular reuse in the form of bash includes such as at line 35 above. Notice also that some of what is included in my-script.sh is also included in my-script.sh (e.g., at line 59). Having these lines in a separate file enables reuse rather than copy and paste.   

Although the idea of making main.sh delete-button-friendly is reasonable and good for those who know bash well enough to understand what they are reading and and who can judge what to delete and what to keep, a user who knows less about what they're reading in main.sh (i.e., me) might be better served by providing a file that can be included whole hog as in line 35.  To give one very simple example, when I first tried to include main.sh into my own scripts via ". main.sh", it took me a bit of time to realize that the emergency line at the end of main.sh terminates execution so that nothing I do after ". main.sh" will ever execute.  

Also, when I'm likely to need the same boilerplate functionality in multiple scripts in the same project, I'd like to not have to copy what I need into each script, which in addition to bloating the code, introduces a maintenance hassle if I make changes to some aspect of the bash3boilerplate and want those changes to benefit all of the scripts that use bash3boilerplate. 

With the workflow demonstrated in the use-case subdirectory, I expect that I'll reuse bootstrap.sh repeatedly without modification.  I further expect that I'll start each new scripting I write by copying and renaming my-script.sh and my-script.sh-usage and keeping everything in my-script.sh the same except toward the bottom, which is a natural place to start any new scripting. And rather than modifying the read block in the middle of every new script I write, I'd like to have in a separate file -- again so that the my-script.sh script itself comes into each new scripting effort very unmodified until near the bottom. 

If this is deemed useful, I'll be glad to write a GETTING_STARTED.md file that describes the intended workflow.  At least for me, I think I would have come up to speed much faster on using all this fabulous technology with the workflow that I'm describing.  It took me quite some time to figure out how to refactor main.sh to make efficient use of it in OpenCoarrays and my very next step will be to incorporate this new workflow into enhancing the OpenCoarrays installation scripts.